docs/guides/platforms/gke/index.md

```diff
@@ -60,591 +60,513 @@ $ gcloud projects add-iam-policy-binding sample-project \
     --role "roles/storage.admin"
 ```
 
-> For GCS backend, if the bucket does not exist, Stash needs `Storage Object Admin` role permissions to create the bucket. For more details, please check the following [guide](/docs/guides/backends/gcs/index.md).
+Also ensure that the required IAM roles for Workload Identity is added.
 
-## Prepare MariaDB 
+## Prepare KubeStash Operator
 
-In this section, we are going to deploy a MariaDB database using KubeDB. Then, we are going to insert some sample data into it.
+In this section, we are going to bind the Kubernetes service account for KubeStash operator and the IAM service account `bucket-accessor` that we have created earlier. This binding allows the Kubernetes service account to act as the IAM service account.
 
-### Deploy MariaDB using KubeDB
-
-At first, let’s deploy a MariaDB standalone database named `sample-mariadb` using KubeDB,
-
-```yaml
-apiVersion: kubedb.com/v1alpha2
-kind: MariaDB
-metadata:
-  name: sample-mariadb
-  namespace: demo
-spec:
-  version: "10.5.8"
-  replicas: 1
-  storageType: Durable
-  storage:
-    storageClassName: "standard"
-    accessModes:
-    - ReadWriteOnce
-    resources:
-      requests:
-        storage: 1Gi
-  terminationPolicy: WipeOut
-```
+We have installed KubeStash in `kubestash` namespace. Let’s add the IAM annotations to the `ServiceAccount`,
 
 ```bash
-$ kubectl apply -f https://github.com/kubestash/docs/raw/{{< param "info.version" >}}/docs/guides/platforms/gke/examples/mariadb.yaml
-mariadb.kubedb.com/sample-mariadb created
+$ kubectl annotate sa -n kubestash kubestash-kubestash-operator iam.gke.io/gcp-service-account="bucket-accessor@sample-project.iam.gserviceaccount.com" 
+serviceaccount/kubestash-kubestash-operator annotated
 ```
 
-Now, wait for the database pod `sample-mariadb-0` to go into Running state,
+Now, let's bind it with the IAM service account,
 
 ```bash
-$ kubectl get pod -n demo sample-mariadb-0
-NAME               READY   STATUS    RESTARTS   AGE
-sample-mariadb-0   1/1     Running   0          29m
+$ gcloud iam service-accounts add-iam-policy-binding bucket-accessor@sample-project.iam.gserviceaccount.com \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:sample-project.svc.id.goog[kubestash/kubestash-kubestash-operator]"
 ```
 
-Once the database pod is in Running state, verify that the database is ready to accept the connections.
+> For Standard cluster, you may need to provide `nodeSelector` during KubeStash installation. Use `--set-string kubestash-operator.nodeSelector."iam\\.gke\\.io/gke-metadata-server-enabled"="true"` during KubeStash installation. For Autopilot clusters, omit the `nodeSelector`. Autopilot rejects this `nodeSelector` because all nodes use Workload Identity.
```

Comment:
For a Standard cluster, you might require a `nodeSelector` for the pod that uses workload identity to connect to the backend. However, for Autopilot clusters, there's no need to set the `nodeSelector`. Autopilot rejects this because all nodes utilize Workload Identity.

To set the `nodeSelector` for the operator pod, you can use the flag `--set-string kubestash-operator.nodeSelector."iam.gke.io/gke-metadata-server-enabled"="true"` when installing or upgrading KubeStash via Helm.